config/config.go

```diff
@@ -8,19 +8,20 @@ import (
 
 // Configuration
 type Configuration struct {
-	ExternalURL     string             `mapstructure:"external_url"`
-	Host            string             `mapstructure:"host"`
-	Port            int                `mapstructure:"port"`
-	AdminPort       int                `mapstructure:"admin_port"`
-	DefaultTimeout  uint64             `mapstructure:"default_timeout_ms"`
-	CacheURL        Cache              `mapstructure:"cache"`
-	RecaptchaSecret string             `mapstructure:"recaptcha_secret"`
-	HostCookie      HostCookie         `mapstructure:"host_cookie"`
-	Metrics         Metrics            `mapstructure:"metrics"`
-	DataCache       DataCache          `mapstructure:"datacache"`
-	StoredRequests  StoredRequests     `mapstructure:"stored_requests"`
-	Adapters        map[string]Adapter `mapstructure:"adapters"`
-	MaxRequestSize  int64              `mapstructure:"max_request_size"`
+	ExternalURL      string             `mapstructure:"external_url"`
+	Host             string             `mapstructure:"host"`
+	Port             int                `mapstructure:"port"`
+	AdminPort        int                `mapstructure:"admin_port"`
+	DefaultTimeout   uint64             `mapstructure:"default_timeout_ms"`
+	CacheURL         Cache              `mapstructure:"cache"`
+	RecaptchaSecret  string             `mapstructure:"recaptcha_secret"`
+	HostCookie       HostCookie         `mapstructure:"host_cookie"`
+	Metrics          Metrics            `mapstructure:"metrics"`
+	DataCache        DataCache          `mapstructure:"datacache"`
+	StoredRequests   StoredRequests     `mapstructure:"stored_requests"`
+	Adapters         map[string]Adapter `mapstructure:"adapters"`
+	MaxRequestSize   int64              `mapstructure:"max_request_size"`
+	AdServerCurrency string             `mapstructure:"ad_server_currency"` // Currency strings must be ISO 4217 alpha codes
```

Comment:
I'd be more comfortable making it a required field. I want to avoid publishers not being aware that there's a default value set in the config. can you first clarify "or just made it a required field and returned a 400 if it didn't exist on requests."? do you mean it has to be set in EITHER the config or request? or it has to be set in the config and in the request it's optional?

Reply:
Sounds good to me. Yes, I meant it has to be set in EITHER the config or the request.

The order of operations for the endpoint are:

- Parse the request for Config IDs
- Merge Config Data into HTTP Request data
- Set any unknown values if we can (e.g. user agent string from HTTP header)
- Run all validation

To make this required, you'd edit the validation phase. The validation code won't know whether some piece of data was Request or Config, because those phases were already over.

Reply:
changes made.